src/sentry/api/endpoints/project_group_index.py

```diff
@@ -226,6 +228,10 @@ def get(self, request, project):
                     pass
                 else:
                     matching_group = Group.objects.get(id=mapping.group_id)
+                    try:
+                        matching_event = Event.objects.get(event_id=query)
```

Comment:
This is incorrect and could be a security hole.

`event_id`'s aren't globally unique, so this query right now is making the assumption that it is, and we will only find one row in the entire database to match.

If we look at our model definition, https://github.com/getsentry/sentry/blob/master/src/sentry/models/event.py#L57

We are unique'd on `project_id, event_id`. This is just saying that `event_id`'s are unique per project scope.

Therefore, we also need to qualify the query here with `project_id`

So:

```python
Event.objects.get(
  project_id=project.id,
  event_id=query,
)
```

Would give us the results we expect.